### ISSUE TITLE
xIEEsc | xUac: Error thrown during MOF compilation since version 1.5.0

### ISSUE THREAD

**Issue by fullenw1** [dsccommunity/xSystemSecurity#34]
#### Details of the scenario you tried and the problem that is occurring
I am already using the module since version 1.4.0.0
I didn't change the configuration, and only upgraded the module to version 1.5.0.
Since then the MOF compilation is throwing an error.
When I change the module version back to 1.4.0 (without any other change), the compilation is working fine again.

#### Verbose logs showing the problem
```powershell
VERBOSE: 	Result found for xSystemSecurity
PSDesiredStateConfiguration\Node : The term 'xIEEsc\xIEEsc' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At New-DscMof.ps1:140 char:5
+     Node $AllNodes.NodeName {
+     ~~~~
    + CategoryInfo          : ObjectNotFound: (xIEEsc\xIEEsc:String) [PSDesiredStateConfiguration\node], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException,PSDesiredStateConfiguration\node
 
Compilation errors occurred while processing configuration 'New-DscConfiguration'. Please review the errors reported in error stream and modify your configuration code appropriately.
    + CategoryInfo          : InvalidOperation: (New-DscConfiguration:String) [], InvalidOperationException
    + FullyQualifiedErrorId : FailToProcessConfiguration
```
Or when I comment out the xIEEsc section I have the following error message:

```powershell
VERBOSE: 	Result found for xSystemSecurity
PSDesiredStateConfiguration\Node : The term 'xUAC\xUac' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At New-DscMof.ps1:140 char:5
+     Node $AllNodes.NodeName {
+     ~~~~
    + CategoryInfo          : ObjectNotFound: (xUAC\xUac:String) [PSDesiredStateConfiguration\node], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException,PSDesiredStateConfiguration\node
 
Compilation errors occurred while processing configuration 'New-DscConfiguration'. Please review the errors reported in error stream and modify your configuration code appropriately.
    + CategoryInfo          : InvalidOperation: (New-DscConfiguration:String) [], InvalidOperationException
    + FullyQualifiedErrorId : FailToProcessConfiguration
```
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
Here is the Datum configuration:

```powershell
xSystemSecurity:
  xIEEsc:
    - UserRole: Administrators
      IsEnabled: False
  xUac:
    Setting: AlwaysNotify
```

#### The operating system the target node is running

NA because the problem appears during the creation of the MOF.

#### Version and build of PowerShell the target node is running

NA because the problem appears during the creation of the MOF.

#### Version of the DSC module that was used

1.5.0

**Comment by johlju** [comment]
Not sure why that happened, the module output looks the same to me. 🤔 Both these resources are pending deprecation, but happy to merge a fix if anyone can debug and send in a PR (before deprecation).

Try the new resources:

The resource xIEEsc has been refactored and merged to the module ComputerManagementDsc as `IEEnhancedSecurityConfiguration`. The resource xUac has also been refactored as `UserAccountControl` and is pending review in the module ComputerManagementDsc. 

IEEnhancedSecurityConfiguration will be part of the v8.0.0-preview0001 in the PowerShell Gallery (hopefully in a few minutes). Here is the release notes so far at least: 
https://github.com/dsccommunity/ComputerManagementDsc/releases/tag/v8.0.0-preview0001

UserAccountControl will be released as soon as it is merged.

Report any findings to the repository https://github.com/dsccommunity/ComputerManagementDsc

**Comment by johlju** [comment]
I can only see two things that could potentially do this. Either this that the below was added to the module manifest in v1.5.0, or the change to semantic version. Can someone verify if removing the below from the module manifest in 1.5.0 solve the compilation issue? 

```
    DscResourcesToExport = @(
        'xIEEsc'
        'xUAC'
        'xFileSystemAccessRule'
    )
```

**Comment by fullenw1** [comment]
I found the reason.
It is the same one as for the latest version of the **ComputerManagementDsc** module.
Those 2 resources are not part of the module published on PSGallery.

```powershell
PS> Get-DscResource -Module xSystemSecurity | Where-Object -Property Version -EQ '1.5.0'

ImplementedAs   Name                      ModuleName                     Version    Properties                                        
-------------   ----                      ----------                     -------    ----------                                        
PowerShell      xFileSystemAccessRule     xSystemSecurity                1.5.0      {Identity, Path, DependsOn, Ensure...}

PS>
```

**Comment by johlju** [comment]
But the resource code is present in the downloaded module right? Could you please also remove the property I mention above from the module manifest and see if finds the resources after that? 🤔

**Comment by johlju** [comment]
So it seems these lines need to be reverted back to `FunctionsToExport = '*'`, or rather `FunctionsToExport = @('xIEESc')` and `FunctionsToExport = @('xUAC')` respectively.

https://github.com/dsccommunity/xSystemSecurity/blob/63ccc0076d58f9f9448eab9f81058679ac846a0a/source/DSCResources/xIEEsc/xIEEsc.psd1#L24

https://github.com/dsccommunity/xSystemSecurity/blob/63ccc0076d58f9f9448eab9f81058679ac846a0a/source/DSCResources/xUAC/xUAC.psd1#L21

**Comment by johlju** [comment]
If someone can send in a PR with this change it would be much appreciated. No unit tests needed.

**Comment by fullenw1** [comment]
@johlju 

Those resources are now available in the ComputerManagementDSC module v8.0.0 which has just been released.
I think we can close the issue :)
